tests/schema/tests.py

```diff
@@ -2906,3 +2909,35 @@ class Meta:
         with connection.schema_editor(atomic=True) as editor:
             editor.alter_db_table(Foo, Foo._meta.db_table, 'renamed_table')
         Foo._meta.db_table = 'renamed_table'
+
+
+@unittest.skipUnless(connection.vendor == 'postgresql', 'PostgreSQL tests')
+class TestSequenceAfterAlterField(TestCase):
+    """
+    #30266 -- altering a field's type should not break fetching the latest
+    sequential value.
+    """
+
+    @override_settings(
+        INSTALLED_APPS=[
+            "schema.test_apps.author_app",
+        ]
+    )
+    def test_sequence_after_alter_field(self):
+        # 1. modify Author primary key to BigInt
+        executor = MigrationExecutor(connection)
+        executor.migrate([
+            ("author_app", "0001_initial"),
+            ("author_app", "0002_alter_id"),
+        ])
+
+        # 2. load fixture with Author pk=1
+        management.call_command(
+            'loaddata',
+            'author',
+            verbosity=0,
+        )
+
+        # 3. Insert new Author, which shouldn't reuse pk=1
+        from .test_apps.author_app.models import Author
+        Author.objects.create(name='some author')
```

Comment:
I think we can add a test to the `schema/tests.py` without creating a separate `author_app`, fixture, migrations etc., e.g.
```python
    def test_alter_implicit_autofield_pk_to_bigautofield_pk(self):
        with connection.schema_editor() as editor:
            editor.create_model(Author)

        old_field = Author._meta.get_field('id')
        new_field = BigAutoField(primary_key=True)
        new_field.set_attributes_from_name('id')
        new_field.model = Author
        with connection.schema_editor() as editor:
            editor.alter_field(Author, old_field, new_field, strict=True)

        Author.objects.create(name='Foo', pk=1)
        with connection.cursor() as cursor:
            cursor.execute(connection.ops.sequence_reset_sql(no_style(), [Author])[0])
        # This will fail if sequence is missing an owner .... some comment
        Author.objects.create(name='Bar')
```

Reply:
Thanks for the simplification.  I've updated the PR with your suggestion.

One thing I didn't take any action on is indicating what release branches this fix should go into.  It seems worth of going into the active ones (1.11, 2.0, etc.). Is there any action I should take for that?

Reply:
@pydolan Thanks for update :+1: I'm sorry but I think that this fix doesn't qualify for a backport based on our [supported versions policy](https://docs.djangoproject.com/en/dev/internals/release-process/#supported-versions).